File: Scripts/Brain.py
from perceptron import Perceptron
class Brain():
    def __init__(self, weights=None):
        self.perceptrons = [Perceptron(4,weights[0]),Perceptron(4,weights[1]),Perceptron(2,weights[2])] if weights else [Perceptron(4),Perceptron(4),Perceptron(2)]
    def chooseState(self,inputs):
        perSum1 = self.perceptrons[0].createSum(inputs)
        perSum2 = self.perceptrons[1].createSum(inputs)
        finalSum = self.perceptrons[2].createSum([perSum1,perSum2])
        output = 0
        if finalSum >= 0:
            output = 1
        else:
            output = -1
        return output
    def getWeights(self):
        weights = []
        weights.append(self.perceptrons[0].getWeights())
        weights.append(self.perceptrons[1].getWeights())
        weights.append(self.perceptrons[2].getWeights())
        return weights

File: Assets/Scripts/NeuralNetwork/Brain.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Brain
{
    private ArrayList perceptrons;
    private float[][] weights;
    public Brain(){
        perceptrons = new ArrayList();
        perceptrons.Add(new Perceptron(4));
        perceptrons.Add(new Perceptron(4));
        perceptrons.Add(new Perceptron(2));
    }
    public Brain(float[][] weights){
        this.weights = weights;
        perceptrons = new ArrayList();
        perceptrons.Add(new Perceptron(4, weights[0]));
        perceptrons.Add(new Perceptron(4, weights[1]));
        perceptrons.Add(new Perceptron(2, weights[2]));
    }
    public int chooseState(float[] inputs){
        Perceptron per1 = (Perceptron)perceptrons[0];
        Perceptron per2 = (Perceptron)perceptrons[1];
        Perceptron per3 = (Perceptron)perceptrons[2];

        float perSum1 = per1.createSum(inputs);
        float perSum2 = per2.createSum(inputs);
        float finalSum = per3.createSum(new float[]{perSum1,perSum2});

        int output = 0;
        if(finalSum >= 0){
            output = 1;
        }
        else{
            output = -1;
        }

        return output;
    }
    public float[][] getWeights(){
        float[][] allWeights = new float[perceptrons.Count][];
        for(int i=0; i<perceptrons.Count; i++){
            Perceptron currentPer = (Perceptron)perceptrons[i];
            allWeights[i] = currentPer.getWeights();
        }
        return allWeights;
    }
}
